lib/models/download.js

```diff
@@ -30,5 +31,18 @@ schema.plugin(scopeChecks);
 schema.plugin(filterByOrg);
 schema.plugin(addCRUDFunctions);
 
+schema.pre('remove', async function removeFile(next) {
+  if (this.upload) {
+    try {
+      const repo = factory(this.upload.repo);
```

Comment:
Clever, hadn't considered that the repo could change over time, but if you have the creds in for more types, then this handles that